amen/feature.py

```diff
@@ -112,6 +136,27 @@ def at(self, time_slices):
             new_features[key] = self[key].at(time_slices)
         return new_features
 
+    def __iter__(self):
+        """
+        Wrapper to avoid making the user deal with parallel lists
+        """
+        key = list(self.keys())[0]
+        length = len(self[key])
+        for i in range(length):
+            res = {}
+            for key, feature in self.items():
+                res[key] = feature.data[feature.name][i]
+            yield res
+
+    def __len__(self):
+        """
+        Wrapper to avoid making the user deal with parallel lists
+        """
+        key = list(self.keys())[0]
+        feature = self[key]
+        return len(feature)
+
+
```

Comment:
These two are a bit weird because we have a dict, but we really want to be able to loop over the lists that are in the dict.  Should we override `__iter__`, or name our own function to do this?